shotgun_api3/shotgun.py

```diff
@@ -1752,8 +1794,24 @@ def _response_errors(self, sg_response):
         """
 
         if isinstance(sg_response, dict) and sg_response.get("exception"):
-            raise Fault(sg_response.get("message",
-                "Unknown Error"))
+            
+            # handle the following cases explicitly
+            #
+            # {'exception': True, 
+            #  'error_code': 102, 
+            #  'message': "Can't authenticate session token 'xxxxxxxx'"}
+            #
+            # {'exception': True, 
+            #  'error_code': 102, 
+            #  'message': "Session token 'xxxxxxxx has expired'"}
+            
+            msg = sg_response.get("message")
+            
+            if msg and ("Can't authenticate session token" in msg or "has expired" in msg):
+                raise SessionAuthenticationFault(msg)
```

Comment:
Just looked at the shotgun code and 102 is ERR_AUTH, so we can be sure that it is a failed authentication.  Is that enough for the use case you have (ie if you get a 102 and were trying to use a session token then you know what happened)?

Other than that... even the vagueness of the check seems bad "has expired" could match something else in the future.

And looking through the possible session errors, there are more:
User associated with session token '#{auth_args['session_token']} no longer authorized.'
Corrupted session token '#{auth_args['session_token']}'

and another that wouldn't be 102 on poorly formatted session token:
API #{mtd}() session_uuid is not a valid UUID argument:\n#{API3Handler.ppstr(auth_args)}

Feels like simply if you get a 102 when trying to authenticate with a session then you know what happened and matching the message shouldn't be needed.